src/Plugin/views/argument_default/Membership.php

```diff
@@ -0,0 +1,174 @@
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Plugin\Context\ContextProviderInterface $og_context
+   *   The OG context provider.
+   * @param \Drupal\og\MembershipManagerInterface $og_membership
+   *   The OG membership manager.
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   The user to be evaluated.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ContextProviderInterface $og_context, MembershipManagerInterface $og_membership, AccountInterface $user) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->ogContext = $og_context;
+    $this->ogMembership = $og_membership;
+    $this->ogUser = $user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('og.context'),
+      $container->get('og.membership_manager'),
+      $container->get('current_user')->getAccount()
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    $options['og_group_membership'] = ['default' => ''];
+
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    $form['og_group_membership'] = [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getArgument() {
+    // Currently restricted to node entities.
+    return implode(',', $this->getCurrentUserGroupIds('node'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheMaxAge() {
+    return Cache::PERMANENT;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    // This cache context is the best thing we have right now.
+    // og_role takes in consideration the user memberships and
+    // the roles held in the corresponding groups, and while it
+    // is one level too granular, i.e. the context will be more
+    // fragmented than strictly needed, it works.
+    return ['og_role'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    $group = $this->getGroup();
+    if ($group instanceof ContentEntityInterface) {
+      $tag = $group->getEntityTypeId() . ':' . $group->id();
+      return Cache::buildTags('og-group-content', [$tag]);
+    }
+
+    return [];
+  }
+
+  /**
+   * Returns groups that current user is a member of.
+   *
+   * @param string $entity_type
+   *   The entity type, defaults to 'node'.
+   *
+   * @return array
+   *   An array of groups, or an empty array if no group is found.
+   */
+  protected function getCurrentUserGroupIds($entity_type = 'node') {
+    $groups = $this->ogMembership->getUserGroupIds($this->ogUser->id());
```

Comment:
+    $groups = $this->ogMembership->getUserGroupIds($this->ogUser->id());
```suggestion
    $groups = $this->ogMembership->getUserGroupIdsByRoleIds($this->ogUser->id(), [$roles]);
```